Otus-DevOps-2018-02/ifqthenp_microservices
# HW 16. Docker networking & Docker Compose (docker-4)

# Completed Homework №16

- :large_blue_diamond: [x] Main tasks
- :large_orange_diamond: [x] Extra tasks for self-study

## The following tasks have been completed

- :large_blue_diamond: Launched number of containers with different Docker network drivers: `none`, `host`, and `bridge`
- :large_blue_diamond: Explored Docker networks functionality using `joffotron/docker-net-tools` container with pre-installed networking utilities
- :large_blue_diamond: Created `front_net` and `back_net` networks for Reddit application. Connected `post` and `comment` containers to both networks
- :large_blue_diamond: Created `docker-compose.yml` file and transposed `docker container` commands with networks and network aliases to Compose format
- :large_blue_diamond: Populated default values referenced in the Compose file using environment variables set in `.env` file
- :large_orange_diamond: Default project name for Compose can be set in `.env` file using Compose environment variable `COMPOSE_PROJECT_NAME`. Alternatively, it can be set using `-p` command-line option
- :large_orange_diamond: Created `docker-compose.override.yml` file to launch Ruby applications on Puma server in debug mode with 2 workers

## How to get project running

- `cd` into `src` folder and run `docker-compose up -d`

This will launch four containers based on alpine images and with Puma servers running in debug mode with two workers.

## How to check that project actually works

- Ensure that the build passed TravisCI tests
  - [#andrei_bogomja](https://devops-team-otus.slack.com/messages/C9M6L1XB3) Slack channel
  - actual GitHub pull request page
- Run `docker-compose ps` and see the output in terminal. It should look similar to the one below

```shell
        Name                     Command             State           Ports
-----------------------------------------------------------------------------------
reddit-app_comment_1   puma --debug -w 2             Up
reddit-app_post_1      python3 post_app.py           Up
reddit-app_post_db_1   docker-entrypoint.sh mongod   Up      27017/tcp
reddit-app_ui_1        puma --debug -w 2             Up      0.0.0.0:9292->9292/tcp
```

## PR checklist
